System/Random.hs

```diff
@@ -564,7 +475,7 @@ getStdGen  = readIORef theStdGen
 
 theStdGen :: IORef StdGen
 theStdGen  = unsafePerformIO $ do
-   rng <- mkStdRNG 0
+   rng <- SM.initSMGen
```

Comment:
I wonder how is that `theStdGen` never had `NOINLINE` pragma on it, I thought ghc warns you about this sort of thing

Reply:
I've added `NOINLINE` now to bring this in line with https://github.com/phadej/splitmix/blob/04e74fbc8447eafeb3181934e5ec581be27fc273/src/System/Random/SplitMix.hs#L291-L293